### PR TITLE
Add prototype

### PR DESCRIPTION
**Pull request by mogottsch**
### What does this PR do?

This PR contains an initial prototype for `helm-values-gen`.
`helm-values-gen` generates a YAML payload that contains all default values in a given JSON schema. 
This is useful when maintaining a helm project with a `values.schema.json` that already contains the default values that are expected in `values.yaml`.
 
### How does it look like?
Preview:

```jsonc
// ./pkg/convert/testdata/simple.json
{
  "$id": "https://example.com/person.schema.json",
  "$schema": "https://json-schema.org/draft/2020-12/schema",
  "properties": {
    "age": {
      "type": "integer",
      "default": 21
    },
    "lastName": {
      "type": "string",
      "default": "Doe"
    },
    "height": {
      "type": "number",
      "default": 1.78
    },
    "isAdult": {
      "type": "boolean",
      "default": true
    }
  },
  "type": "object"
}

```

```
go run ./main.go ./pkg/convert/testdata/simple.json
age: 21
height: 1.78
isAdult: true
lastName: Doe
```

### Any background context you can provide?

Towards https://github.com/giantswarm/roadmap/issues/1983 

### What is needed from the reviewers?

There are a couple of test files you can try out in `./pkg/convert/testdata/`.

### Do the docs need to be updated?

There are no docs yet.

### Should this change be mentioned in the release notes?

- [x] CHANGELOG.md has been updated (if it exists)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,30 +1,119 @@
...
+# helm-values-gen
+
+`helm-values-gen` generates a YAML payload that contains all default values in
+a given JSON schema. This is useful when maintaining a helm project with a
+`values.schema.json` that already contains the default values that are
+expected in `values.yaml`.
+
+## Installation
+
+Currently, `helm-values-gen` can be installed as go binary.
```
Comment (marians) [inline README.md:10]:
```suggestion
Currently, `helm-values-gen` can be installed as Go binary.
```

File: README.md [hunk 2]
```diff
@@ -1,30 +1,119 @@
...
+      "type": "array"
+    },
+    "lastName": {
+      "type": "string",
+      "default": "Doe"
+    }
+  },
+  "type": "object"
+}
+```
```
Comment (marians) [inline README.md:71]: I suggest to order the `type` key to the top, to improve readability

File: pkg/convert/testdata/address.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{
+  "$id": "https://example.com/person.schema.json",
```
Comment (marians) [inline pkg/convert/testdata/address.json:2]: Do we really need these `$id` fields?

File: README.md [hunk 4]
```diff
@@ -1,30 +1,119 @@
...
+In the future it will be (additionally) possible to install it as a helm plugin.
+
+```nohighlight
+go install github.com/giantswarm/helm-values-gen@latest
+```
+
+## Usage
+
+Executing `schemalint verify` without any options will check whether a file is
+valid JSON Schema and whether it is normalized.
```
[the remark this answers is not in the document]
Comment (mogottsch) [reply README.md:20]: Oh no I got caught copy pasting :laughing: 
I removed it.

**Comment by marians** [comment]
Should there be a simple test case where one property does not have a default? This would make it easy to spot that the property is omitted from the YAML generation.

File: pkg/convert/testdata/address.json (hunk 3 above)
Comment (mogottsch) [reply pkg/convert/testdata/address.json:2]: No, I removed them.

File: README.md (hunk 2 above)
Comment (mogottsch) [reply README.md]: Done

**Comment by mogottsch** [comment]
> Should there be a simple test case where one property does not have a default? This would make it easy to spot that the property is omitted from the YAML generation.

I've added a test.

**Review by marians — APPROVED** [review]

**Pull request merged**